pkg/engine/plan.go

```diff
@@ -242,6 +242,14 @@ func (res *planResult) Walk(ctx *Context, events deploy.Events, preview bool) (d
 		rst, err = resource.StatusOK, nil
 	}()
 
+	// Asynchronously listen for cancellation, and deliver that signal to plan.
+	go func() {
+		select {
```

Comment:
Unless I'm mistaken, that would cause this function to exit, which I think we _don't_ want, right? The semantics of this function are to exit on `Terminate`, no?

Reply:
This function will exit anyway the first time someone writes to `ctx.Cancel.Canceled()`, is that not what you want?

Reply:
Oh, I see, sorry, I thought (on my phone) you were suggesting to move this to the select below. Yes, this makes sense. I don't have a good sense of idioms in Go yet...

Reply:
Actually I think we want to do something like this, to avoid leaking the goroutine:

```
	// Asynchronously listen for cancellation, and deliver that signal to plan.
	go func() {
		select {
		case <-ctx.Cancel.Canceled():
			res.Plan.SignalCancellation()
		case <-done:
			return
		}
	}()
```

Reply:
Sounds good to me. I don't think we care much about leaking goroutines since we're generally exiting pretty quickly.